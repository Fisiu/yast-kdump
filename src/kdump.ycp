/* ------------------------------------------------------------------------------
 * Copyright (c) 2006 Novell, Inc. All Rights Reserved.
 *
 *
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of version 2 of the GNU General Public License as published by the
 * Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, contact Novell, Inc.
 *
 * To contact Novell about this file by physical or electronic mail, you may find
 * current contact information at www.novell.com.
 * ------------------------------------------------------------------------------
 */

/**
 * File:	clients/kdump.ycp
 * Package:	Configuration of kdump
 * Summary:	Main file
 * Authors:	Jozef Uhliarik <juhliarik@suse.com>
 *
 * $Id: kdump.ycp 27914 2006-02-13 14:32:08Z locilka $
 *
 * Main file for kdump configuration. Uses all other files.
 */

{

/***
 * <h3>Configuration of kdump</h3>
 */

textdomain "kdump";

/* The main () */
y2milestone ("----------------------------------------");
y2milestone ("Kdump module started");

import "Progress";
import "Report";
import "Summary";

import "CommandLine";
include "kdump/wizards.ycp";

map cmdline_description = $[
    "id" 	: "kdump",
    /* Command line help text for the Xkdump module */
    "help"	: _("Configuration of kdump"),
    "guihandler"        : KdumpSequence,
    "initialize"        : Kdump::Read,
    "finish"            : Kdump::Write,
    "actions"           : $[
	// FIXME TODO: fill the functionality description here
    ],
    "options"		: $[
	// FIXME TODO: fill the option descriptions here
    ],
    "mappings"		: $[
	// FIXME TODO: fill the mappings of actions and options here
    ]
];

/* is this proposal or not? */
boolean propose = false;
list args = WFM::Args();
if(size(args) > 0) {
    if(is(WFM::Args(0), path) && WFM::Args(0) == .propose) {
        y2milestone("Using PROPOSE mode");
        propose = true;
    }
}

/* main ui function */
any ret = nil;

if(propose) ret = KdumpAutoSequence();
else ret = CommandLine::Run(cmdline_description);
y2debug("ret=%1", ret);

/* Finish */
y2milestone("Kdump module finished");
y2milestone("----------------------------------------");

return ret;

/* EOF */
}
