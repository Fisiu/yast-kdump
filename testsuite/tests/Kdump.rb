# encoding: utf-8

module Yast
  class KdumpClient < Client
    def main
      # testedfiles: Kdump.ycp

      Yast.include self, "testsuite.rb" 
      # map I_READ = $[];
      # map I_WRITE = $[];
      # map I_EXEC = $[
      # 	"target" : $[
      # 	    "bash_output" : $[],
      # 	],
      #     ];
      #
      # TESTSUITE_INIT([I_EXEC, I_READ, I_WRITE], nil);
      #
      #
      # import "Kdump";
      #
      #
      # DUMP("Kdump::Modified");
      # TEST(``(Kdump::Modified()), [], nil);

      nil
    end
  end
end

Yast::KdumpClient.new.main
