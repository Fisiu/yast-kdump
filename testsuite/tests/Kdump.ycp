{

// testedfiles: Kdump.ycp

include "testsuite.ycp";
TESTSUITE_INIT([], nil);

import "Kdump";

DUMP("Kdump::Modified");
TEST(``(Kdump::Modified()), [], nil);

}
