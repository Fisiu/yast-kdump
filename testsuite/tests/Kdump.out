Dump	Kdump::Modified
Return	false
